frontend/src/widgets/Navbar/Navbar.tsx

```diff
@@ -1,54 +1,102 @@
 import { Typography } from "@mui/material";
-import { type FC, memo, useState } from "react";
+import { type FC, memo, useCallback, useEffect, useState } from "react";
 import {
-  CustomIcon,
+  CustomLink,
+  CustomLoginIcon,
+  CustomLogoutIcon,
   CustomTab,
   CustomTabs,
   Header,
+  LoginButton,
   LogoutButton,
 } from "./Navbar.style";
-import { RoutePaths } from "../../shared";
-import { Link } from "react-router";
+import {
+  navbarNoTabsRoutes,
+  navbarStatisticsActiveTabRoutes,
+  RoutePaths,
+} from "../../shared";
+import { Link, useLocation } from "react-router";
 
 enum NavbarTabsEnum {
   STATISTICS_PAGE,
   TEAM_PAGE,
 }
 
-export const Navbar: FC = memo(() => {
+type NavbarProps = {
+  isAuth: boolean;
+};
+
+export const Navbar: FC<NavbarProps> = memo(({ isAuth }) => {
+  const { pathname } = useLocation();
+
+  const setActiveTabFromPathname = useCallback(() => {
+    return !navbarStatisticsActiveTabRoutes.includes(pathname)
+      ? NavbarTabsEnum.TEAM_PAGE
+      : NavbarTabsEnum.STATISTICS_PAGE;
+  }, [pathname]);
+
```

Comment:
```suggestion

const getCurrentTab = (pathname: string) => navbarStatisticsActiveTabRoutes.includes(pathname)
    ? NavbarTabsEnum.STATISTICS_PAGE
    : NavbarTabsEnum.TEAM_PAGE;

export const Navbar: FC<NavbarProps> = memo(({ isAuth }) => {
  const { pathname } = useLocation();
  const [activeTab, setActiveTab] = useState<NavbarTabsEnum>(getCurrentTab(pathname));

  useEffect(() => {
    setActiveTab(getCurrentTab(pathname));
  }, [pathname])

  const handleTabChange = (_: unknown, newValue: NavbarTabsEnum) => setActiveTab(newValue);


```

useCallback тут лишний